src/TTextEdit.cpp

```diff
@@ -784,53 +785,43 @@ void TTextEdit::highlightSelection()
             break;
         }
 
-        for (;; ++currentX) {
+        for (;currentX < maxX; ++currentX) {
             if ((currentY == mPB.y()) && (currentX > mPB.x())) {
                 break;
             }
-            if (currentX < static_cast<int>(mpBuffer->buffer.at(currentY).size())) {
-                if (!(mpBuffer->buffer.at(currentY).at(currentX).isSelected())) {
-                    mpBuffer->buffer.at(currentY).at(currentX).select();
-                    mpBuffer->dirty[currentY] = true;
-                }
-            } else {
-                break;
+            if (!(mpBuffer->buffer.at(currentY).at(currentX).isSelected())) {
+                mpBuffer->buffer.at(currentY).at(currentX).select();
+                mpBuffer->dirty[currentY] = true;
             }
         }
     }
 
     update(mSelectedRegion.boundingRect());
+    update(newRegion);
     mSelectedRegion = newRegion;
 }
 
 void TTextEdit::unHighlight()
 {
-    int y1 = mPA.y();
+    int y1;
+
+    normaliseSelection();
```

Comment:
You could just put this before a `int y1 = mPA.y()` there is no need to separate declaring `y1` and assigning a value to it is there?

Reply:
You need to separate declaring y1 and assigning it because normalizeSelection() assigns mPA, which is then assigned to y1.

Reply:
Yeah but `y1` is a local variable which is not changed by the call to `normaliseSelection()` and it isn't used before that is called so it is fine to have:
```C++
void TTextEdit::unHighlight()
{
    normaliseSelection();
    int y1 = mPA.y();

// ...
```

Reply:
You are absolutely correct. Maybe I got caught in my old habit of declaring variables at the beginning of functions.